AXI DMAC: Add Framelock, AutoRun, fsync, tlast (#1332)
analogdevicesinc/hdl: docs/library/axi_dmac/index.rst

chained descriptor transfers as a single transfer, thus improving the performance
of the application.

External Synchronization
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

This feature allows external components to throttle the consumption of
descriptors queued by the software. A transfer will start only after the
assertion of the external sync signal for at least one clock cycle.

The sync signal can be either in source or destination clock domain or both.

> This feature does not ensures fixed latency from the assertion of external sync

Reviewer: ```suggestion
This feature does not ensure fixed latency from the assertion of external sync
```

signal and the availability of the data at the destination interface.

Framelock Synchronization
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> This feature adds support for multiple 2D frame buffers which are used in a
> cyclic way. On the same set of buffers a second DMAC core can operate.
> The "Framelock" mechanism ensures no buffer is accessed by two DMACs in the same time.

Reviewer: ```suggestion
This feature adds support for multiple 2D frame buffers, which are used in a
cyclic way. On the same set of buffers, a second DMAC core can operate.
The "Framelock" mechanism ensures no buffer is accessed by two DMACs at the same time.
```

The core can operate in two modes:

> * Writer mode - available in s2mm configuration, the writer DMAC will always
>   skip the current in use readers buffer.
> * Reader mode - available in mm2s configuration, the reader DMAC will stay
>   behind the writers buffer by either repeating or skipping buffers according to

Reviewer: ```suggestion
* Writer mode - available in s2mm configuration, the writer DMAC will always
  skip the current in-use reader's buffer.
* Reader mode - available in mm2s configuration, the reader DMAC will stay
  behind the writer's buffer by either repeating or skipping buffers according to
```

  the speed relationship of the two cores.

The writer and reader DMAC cores must be connected through the dedicated
"framelock" interface. They must be programmed with similar settings regarding
the buffers size, start address and stride through the ``FRAMELOCK_CONFIG`` and
``FRAMELOCK_STRIDE`` registers.

Notice that the reader DMA will start to read the frames only after the writer
finished to store in the DDR at least ``FRAMELOCK_CONFIG_DISTANCE+1`` frames.
This means that while the FRAMELOCK_CONFIG_DISTANCE+1 frames are written into the

> memory the reader DMA won’t output anything.

Reviewer: ```suggestion
memory, the reader DMA won’t output anything.
```
